packages/product-service/src/handlers/getProductById.ts

```diff
@@ -0,0 +1,64 @@
+import { APIGatewayProxyHandler } from "aws-lambda";
+import "source-map-support/register";
+
+import { getProductById } from "src/services";
+
+export const getProductByIdHandler: APIGatewayProxyHandler = async (
+  event,
+  _context
+) => {
+  const { productId } = event.pathParameters;
+  const headers = {
+    "Access-Control-Allow-Origin": "*",
+  };
+
+  try {
+    const product = await getProductById(productId);
+
+    if (product) {
+      const statusCode = 200;
+      return {
+        statusCode,
+        headers,
+        body: JSON.stringify(
+          {
+            message: "Success",
+            statusCode,
+            product,
+          },
+          null,
+          2
+        ),
+      };
+    } else {
+      const statusCode = 404;
+      return {
+        statusCode,
+        headers,
+        body: JSON.stringify(
+          {
+            message: "Product Not Found",
+            statusCode,
+          },
+          null,
+          2
+        ),
+      };
+    }
+  } catch (err) {
+    const statusCode = 400;
```

Comment:
If you have a DB connection failure or sth like this, we will get into this `catch` block, but we should return `500`. 

We could return `400` in the `try` block when validation fails.